cinn/hlir/pe/transform.cc

```diff
@@ -168,6 +168,53 @@ std::vector<Tensor> MatmulV2(const Tensor& A,
   return {temp_res, packedB};
 }
 
+std::vector<Tensor> MatmulMKL(const Tensor& A,
+                              const Tensor& B,
+                              bool trans_a,
+                              bool trans_b,
+                              float alpha,
+                              const std::string& name,
+                              const common::Target& target) {
+  CHECK(target.arch == Target::Arch::X86) << "mkl should be used in the cpu environment";
+  std::vector<Expr> shape_A = A->shape;
+  std::vector<Expr> shape_B = B->shape;
+  int a_dim                 = shape_A.size();
+  int b_dim                 = shape_B.size();
+  CHECK(a_dim == 3U || a_dim == 2U) << "tensor_A's dim should be 2 or 3";
```

Comment:
CHECK(a_dim == 3U || a_dim == 2U) << "tensor_A's dim should be 2 or 3 but get " << a_dim;